syntax = "proto3";

package benchmark.proto;

// Message representing file metadata
message FileMetadataProto {
    string name = 1;
    string path = 2;
    uint64 size = 3;
    uint64 created_at = 4;
    uint64 last_modified = 5;
    repeated string tags = 6;
    uint32 permissions = 7;
    string owner = 8;
    string group = 9;
}

// Message representing a file block
message FileBlockProto {
    string block_id = 1;
    uint64 offset = 2;
    bytes data = 3;
    uint32 checksum = 4;
}